### ISSUE TITLE
Regression: no-unused-vars fails on for blocks with let declarations

### ISSUE THREAD

**Issue by mzgol** [eslint/eslint#1733]
The following file:
```js
(function () {
    "use strict";
    const box = {a: 2};

    for (let prop in box) {
        box[prop] = parseInt(box[prop]);
    }
})();
```
fails with the following `.eslintrc`:
```json
{
    "ecmaFeatures": {
        "blockBindings": true
    }
}
```
with a message:
```
app/modules/components/base/item-panel/page-view/pdf/pdf.js
  5:13  error  prop is defined but never used  no-unused-vars

✖ 1 problem
```

ESLint 0.13.0. It used to work in 0.12.0.

**Comment by nzakas** [comment]
Related to the escope upgrade. In 0.12.0, this wasn't technically working correctly because it was applying the same scoping rules to `var` and `let`. Most likely a bug in escope.

**Comment by nzakas** [comment]
Reported to escope: https://github.com/estools/escope/issues/33#issuecomment-72323191

**Comment by nzakas** [comment]
Found a workaround that doesn't involve fixing escope.
